common/pom.xml

```diff
@@ -37,6 +67,10 @@
       <scope>compile</scope> <!-- override the 'test' scope defined at parent pom.xml -->
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>org.jctools</groupId>
+      <artifactId>jctools-core</artifactId>
+    </dependency>
```

Comment:
but if I have to guess deeper I think the generated pom happens before reduced pom and that's where shade plugin does its magic making anything else that happen after that be affected by it.